### PR TITLE
Implement full dashboard

### PR DESCRIPTION
**Pull request by nam-t24**
This contains all up to date develop commit along with all dashboard changes/updates. Last thing to do it make the number of spotlight events dynamic which will be a simple fix done soon.

### CODE REVIEW

**Comment by WillieCubed** [comment]
#### 📝 Changed routes:


| [**…ents/AnnouncementCards**](https://hackportal-1e5cxa2wl-acmutd.vercel.app/dashboard/Components/AnnouncementCards) | [**…rd/Components/DocLinks**](https://hackportal-1e5cxa2wl-acmutd.vercel.app/dashboard/Components/DocLinks) |
|:-:|:-:|
|<a href="https://hackportal-1e5cxa2wl-acmutd.vercel.app/dashboard/Components/AnnouncementCards">Screenshot of /dashboard/Components/AnnouncementCards</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/a4bz7yVmI79fhxR7mtg27mLzputueVhF2eZh9iCuU10qFcmCsAjkrwRCYwn15jZRRVWVq02PLRoQbS9LaX_raw.png">(view full size)</a> |<a href="https://hackportal-1e5cxa2wl-acmutd.vercel.app/dashboard/Components/DocLinks">Screenshot of /dashboard/Components/DocLinks</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/qhRvq3gMlEZO0U0KSo6wy3z_-oYTo_azp7ec1xoqh80KqqLkPCqf9EgFkP9MIl9CwtCmjQiDCQJpniI9j-sAYw.png">(view full size)</a> |



| [**…Components/MentorCard1**](https://hackportal-1e5cxa2wl-acmutd.vercel.app/dashboard/Components/MentorCard1) | [**…Components/MentorCard3**](https://hackportal-1e5cxa2wl-acmutd.vercel.app/dashboard/Components/MentorCard3) |
|:-:|:-:|
|<a href="https://hackportal-1e5cxa2wl-acmutd.vercel.app/dashboard/Components/MentorCard1">Screenshot of /dashboard/Components/MentorCard1</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/QXoAYT7wGLM9iQoZ5B-HHaTD2o4NzDT0ykLfsSyJI8KXub5iHmBU4wmAW4sre7_EjXft6fsde9GHEn0ZYyyTTQ.png">(view full size)</a> |<a href="https://hackportal-1e5cxa2wl-acmutd.vercel.app/dashboard/Components/MentorCard3">Screenshot of /dashboard/Components/MentorCard3</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/LAeh2Wx0YcdPXRhJfxtYlcGM_RaeU6xzj2gpLuXr3fB6-0IC_Yp9R7e9FOpNltRrUW3MFdOvwKZ1p6YYCjUoRg.png">(view full size)</a> |



| [**…ard/Components/Sidebar**](https://hackportal-1e5cxa2wl-acmutd.vercel.app/dashboard/Components/Sidebar) | [**…mponents/SpotlightCard**](https://hackportal-1e5cxa2wl-acmutd.vercel.app/dashboard/Components/SpotlightCard) |
|:-:|:-:|
|<a href="https://hackportal-1e5cxa2wl-acmutd.vercel.app/dashboard/Components/Sidebar">Screenshot of /dashboard/Components/Sidebar</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/FBlrD2pDJROxc6oPgzCSgN_8DdwUlm0eySd9M9_JW9WMPH334n6jM9HXS4tMAnmdN1yjKPUG-G8q9HgM0yfTqQ.png">(view full size)</a> |<a href="https://hackportal-1e5cxa2wl-acmutd.vercel.app/dashboard/Components/SpotlightCard">Screenshot of /dashboard/Components/SpotlightCard</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/0TQiOD1PyRaZe-eb0zKZUKDzBikWgs_aumgM8LTsoK8GLxTdFvMXfN4Por49dnx_xx4bB1HIhgpcjtYZBwnRfQ.png">(view full size)</a> |



And 5 other routes:
- [**/dashboard/Components/SpotlightCardScroll**](https://hackportal-1e5cxa2wl-acmutd.vercel.app/dashboard/Components/SpotlightCardScroll)
- [**/dashboard/hackerpack**](https://hackportal-1e5cxa2wl-acmutd.vercel.app/dashboard/hackerpack)
- [**/dashboard**](https://hackportal-1e5cxa2wl-acmutd.vercel.app/dashboard)
- [**/dashboard/scan-in**](https://hackportal-1e5cxa2wl-acmutd.vercel.app/dashboard/scan-in)
- [**/dashboard/submit**](https://hackportal-1e5cxa2wl-acmutd.vercel.app/dashboard/submit)



Commit 54a0a45c4295472edc34235ff495672e68cd789d (https://hackportal-1e5cxa2wl-acmutd.vercel.app).

File: pages/dashboard/Components/Sidebar.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { useUser } from '../../../lib/profile/user-data';
+import { useAuthContext } from '../../../lib/user/AuthContext';
+
+function Sidebar() {
+  const { isSignedIn } = useAuthContext();
+  const user = useUser();
+  const role = user.permissions?.length > 0 ? user.permissions[0] : '';
+  return (
```
Comment (abdullah50053) [inline pages/dashboard/Components/Sidebar.tsx:9]: Minor issue with the sidebar on mobile: the text spills over the divider. The divider should be removed completely on mobile.
Also, the user type should be under the name, not to the right of it.
Screen Shot 2021-11-06 at 11 48 00 AM
Screen Shot 2021-11-06 at 11 46 30 AM

**Comment by WillieCubed** [comment]
Deployment logs:
<html>
<body>


Cloning github.com/acmutd/hackportal (Branch: feat/UI_Update, Commit: e525df2)
--
20:12:27.943 | Cloning completed: 543.646ms
20:12:27.976 | Analyzing source code...
20:12:28.990 | Installing build runtime...
20:12:31.676 | Build runtime installed: 2.686s
20:12:33.689 | Looking up build cache...
20:12:37.996 | Build cache downloaded [73.52 MB]: 3847.682ms
20:12:38.735 | Installing dependencies...
20:12:38.742 | Detected `package-lock.json` generated by npm 7...
20:12:41.541 |  
20:12:41.541 | > hackportal@1.0.0 prepare
20:12:41.541 | > husky install
20:12:41.541 |  
20:12:41.621 | husky - Git hooks installed
20:12:41.631 |  
20:12:41.631 | added 3 packages in 3s
20:12:41.631 |  
20:12:41.631 | 132 packages are looking for funding
20:12:41.631 | run `npm fund` for details
20:12:41.642 | Detected Next.js version: 11.1.2
20:12:41.650 | Running "npm run build"
20:12:41.925 |  
20:12:41.925 | > hackportal@1.0.0 build
20:12:41.926 | > next build
20:12:41.926 |  
20:12:43.226 | info  - Using webpack 5. Reason: Enabled by default https://nextjs.org/docs/messages/webpack5
20:12:43.899 | info  - Checking validity of types...
20:12:55.793 |  
20:12:55.794 | ./pages/about/questions.tsx
20:12:55.794 | 124:6  Warning: React Hook useEffect has a missing dependency: 'getAllQuestions'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps
20:12:55.795 |  
20:12:55.795 | ./pages/admin/users.tsx
20:12:55.795 | 37:6  Warning: React Hook useEffect has a missing dependency: 'userData'. Either include it or remove the dependency array. If 'setUsers' needs the current value of 'userData', you can also switch to useReducer instead of useState and read 'userData' in the reducer.  react-hooks/exhaustive-deps
20:12:55.796 | 40:13  Warning: Assignments to the 'timer' variable from inside React Hook useEffect will be lost after each render. To preserve the value over time, store it in a useRef Hook and keep the mutable value in the '.current' property. Otherwise, you can move this variable directly inside useEffect.  react-hooks/exhaustive-deps
20:12:55.796 |  
20:12:55.796 | ./pages/auth/signOut.tsx
20:12:55.797 | 21:6  Warning: React Hook React.useEffect has a missing dependency: 'signOut'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps
20:12:55.798 |  
20:12:55.798 | ./pages/dashboard/index.tsx
20:12:55.799 | 52:6  Warning: React Hook useEffect has a missing dependency: 'props.announcements'. Either include it or remove the dependency array. If 'setAnnouncements' needs the current value of 'props.announcements', you can also switch to useReducer instead of useState and read 'props.announcements' in the reducer.  react-hooks/exhaustive-deps
20:12:55.799 |  
20:12:55.799 | ./pages/profile.tsx
20:12:55.799 | 25:6  Warning: React Hook React.useEffect has a missing dependency: 'checkRedirect'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps
20:12:55.799 | 31:6  Warning: React Hook React.useEffect has a missing dependency: 'router'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps
20:12:55.799 |  
20:12:55.799 | ./pages/register.tsx
20:12:55.800 | 28:6  Warning: React Hook useEffect has a missing dependency: 'checkRedirect'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps
20:12:55.800 |  
20:12:55.800 | info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/basic-features/eslint#disabling-rules
20:12:55.800 | info  - Creating an optimized production build...
20:12:55.856 | > [PWA] Compile client (static)
20:12:55.858 | > [PWA] Auto register service worker with: /vercel/path0/node_modules/next-pwa/register.js
20:12:55.858 | > [PWA] Service worker: /vercel/path0/public/sw.js
20:12:55.858 | > [PWA]   url: /sw.js
20:12:55.858 | > [PWA]   scope: /
20:12:56.068 | > [PWA] Compile server
20:13:00.521 |  
20:13:00.521 | warn - You have enabled the JIT engine which is currently in preview.
20:13:00.521 | warn - Preview features are not covered by semver, may introduce breaking changes, and can change at any time.
20:13:18.387 | info  - Compiled successfully
20:13:18.388 | info  - Collecting page data...
20:13:19.587 | info  - Generating static pages (0/20)
20:13:20.146 | info  - Generating static pages (5/20)
20:13:20.156 |  
20:13:20.156 | Error occurred prerendering page "/dashboard/Components/SpotlightCardScroll". Read more: https://nextjs.org/docs/messages/prerender-error
20:13:20.157 | TypeError: Cannot read property 'substring' of undefined
20:13:20.157 | at SpotlightCardScroll (/vercel/path0/.next/server/chunks/2973.js:59:69)
20:13:20.157 | at d (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:33:498)
20:13:20.157 | at bb (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:36:16)
20:13:20.157 | at a.b.render (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:42:43)
20:13:20.157 | at a.b.read (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:41:83)
20:13:20.158 | at Object.exports.renderToString (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:52:138)
20:13:20.158 | at Object.renderPage (/vercel/path0/node_modules/next/dist/server/render.js:596:45)
20:13:20.158 | at Function.getInitialProps (/vercel/path0/.next/server/pages/_document.js:601:19)
20:13:20.158 | at Object.loadGetInitialProps (/vercel/path0/node_modules/next/dist/shared/lib/utils.js:69:29)
20:13:20.158 | at renderDocument (/vercel/path0/node_modules/next/dist/server/render.js:609:48)
20:13:20.173 | info  - Generating static pages (10/20)
20:13:20.206 | info  - Generating static pages (15/20)
20:13:20.805 | info  - Generating static pages (20/20)
20:13:20.807 |  
20:13:20.808 | > Build error occurred
20:13:20.811 | Error: Export encountered errors on following paths:
20:13:20.811 | /dashboard/Components/SpotlightCardScroll
20:13:20.811 | at /vercel/path0/node_modules/next/dist/export/index.js:487:19
20:13:20.811 | at runMicrotasks (<anonymous>)
20:13:20.811 | at processTicksAndRejections (internal/process/task_queues.js:95:5)
20:13:20.812 | at async Span.traceAsyncFn (/vercel/path0/node_modules/next/dist/telemetry/trace/trace.js:60:20)
20:13:20.812 | at async /vercel/path0/node_modules/next/dist/build/index.js:833:17
20:13:20.812 | at async Span.traceAsyncFn (/vercel/path0/node_modules/next/dist/telemetry/trace/trace.js:60:20)
20:13:20.812 | at async /vercel/path0/node_modules/next/dist/build/index.js:707:13
20:13:20.812 | at async Span.traceAsyncFn (/vercel/path0/node_modules/next/dist/telemetry/trace/trace.js:60:20)
20:13:20.812 | at async Object.build [as default] (/vercel/path0/node_modules/next/dist/build/index.js:77:25)
20:13:20.894 | Error: Command "npm run build" exited with 1


</body>
</html>

**Comment by nam-t24** [comment]
Outdated and changes will be implemented by feat/UI_Update

**Pull request closed without merging**